pkg/template/jsonpath_test.go

```diff
@@ -0,0 +1,144 @@
+package template_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/triggers/pkg/template"
+)
+
+var objects = `{"a":"v","c":{"d":"e"},"empty": "","null": null, "number": 42}`
+var arrays = `[{"a": "b"}, {"c": "d"}, {"e": "f"}]`
+// Checks that we print JSON strings when the JSONPath selects
+// an array or map value and regular values otherwise
+func TestParseJSONPath(t *testing.T) {
+	var objectBody = fmt.Sprintf(`{"body":%s}`, objects)
+	tests := []struct {
+		name  string
+		expr string
+		in     string
```

Comment:
Instead of encoding the object body to a string then unmarshalling it out, would it be easier to make this interface{} and set the fields as maps?

Reply:
yeah, don't have a strong opinion on this -> might be a bit more verbose for the array and objects that contain other embedded objects?